### DISCUSSION TITLE
About Dynamic Routes

### DISCUSSION THREAD

**Discussion by yanqc1996** [vercel/next.js#24843] [Help]
I have encountered a requirement now that I need to implement common/tag_id and common/group_id to render different pages. Only the id is dynamically changed. If dynamic routing is used, how can I implement this function? Thank you for your help

**Comment by gomflo** [comment]
You can create two routes: `common/tag/:id` and `common/group/:id`

**Reply by yanqc1996** [reply]
I mean, I want to achieve matching such as the distinction between common/tag_123 and common/group_123, instead of common/tag/123 and common/group/123, can I distinguish the former routing based on creating different file names?

**Comment by ghost** [comment]
You can use router to get `id` from query string. If you can differentiate between a `tag_id` and a `group_id`, create a conditional statement and render the page accordingly.

```jsx
const tag_id = [1, 3, 5, 7, 9]
const group_id = [2, 4, 6, 8, 0]

const Common = () => {
  const router = useRouter()
  const { id } = router.query

  if (tag_id.includes(id))
    return <TagComponent />

  if (group_id.includes(id))
    return <GruopComponent />
}
```

**Reply by yanqc1996** [reply]
Thank you for your answer. My situation is not the same as this one. The id is not fixed and limited, and there will be more prefix types. This provides me with an idea to distinguish and verify the prefix in the page routing. But I Still want to know, can it be achieved by creating different files based on the dynamic routing api provided by Next.js?

**Reply by ghost** [reply]
From what I know, you can only create a single `[slug]` file for a directory. Another way could be using controller logic to route. Finally, without knowing which `id` is for `tag` or which for `group` you can't route to different pages.

**Reply by gomflo** [reply]
If you will have unlimited prefixes i think the only way is to use **Catch all routes** https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes

**Reply by yanqc1996** [reply]
It should be. In this case, I can only match all routes and write methods to distinguish these routes. It seems that there is no direct way to distinguish this kind of semi-dynamic routes by creating file names.
